app/components/primer/alpha/segmented_control.pcss

```diff
@@ -46,13 +46,15 @@
   }
 
   /* Separator lines */
-  &::before {
-    position: absolute;
-    inset: 0 0 0 -1px;
-    margin-top: var(--control-medium-paddingBlock, 6px);
-    margin-bottom: var(--control-medium-paddingBlock, 6px);
-    content: '';
-    border-left: var(--borderWidth-thin, max(1px, 0.0625rem)) solid var(--color-border-default);
+  &:not(:first-child) {
```

Comment:
The increase in specificity here created a small bug 
<img width="227" alt="image" src="https://user-images.githubusercontent.com/54012/235516788-16360e7c-5666-47dd-9d77-fe80b84c2b6c.png">

If you change line 31, to `border-color: transparent !important;` that should fix it

Reply:
Thanks @jonrohan! I pushed this requested fix 😊